reconcile/skupper_network/reconciler.py

```diff
@@ -0,0 +1,269 @@
+import logging
+from collections.abc import Iterable
+from typing import Any
+
+from sretoolbox.utils import threaded
+
+import reconcile.openshift_base as ob
+from reconcile.skupper_network.models import SkupperSite
+from reconcile.skupper_network.site_controller import LABELS as SITE_CONTROLLER_LABELS
+from reconcile.skupper_network.site_controller import get_site_controller
+from reconcile.utils.oc import OC_Map
+from reconcile.utils.openshift_resource import OpenshiftResource as OR
+from reconcile.utils.openshift_resource import ResourceInventory
+
+
+def delete_skupper_site(
+    site: SkupperSite,
+    oc_map: OC_Map,
+    dry_run: bool,
+    integration_managed_kinds: Iterable[str],
+) -> None:
+    """Delete all skupper resources (leftovers not covered by ResourceInventory) starting with 'skupper-' in a namespace."""
+    oc = oc_map.get_cluster(site.cluster.name)
+    to_delete: dict[str, dict[str, Any]] = {}
+
+    for kind in integration_managed_kinds:
+        # delete everything labeled by us
+        to_delete.update(
+            {
+                f'{item["kind"]}-{item["metadata"]["name"]}': item
+                for item in oc.get_items(
+                    kind=kind,
+                    namespace=site.namespace.name,
+                    labels=SITE_CONTROLLER_LABELS,
+                )
+            }
+        )
+        # delete everything else that starts with 'skupper-'
+        to_delete.update(
+            {
+                f'{item["kind"]}-{item["metadata"]["name"]}': item
+                for item in oc.get_items(kind=kind, namespace=site.namespace.name)
+                if item["metadata"]["name"].startswith("skupper-")
+            }
+        )
```

Comment:
what kind of objects are these? objects created by the deployed skupper?

Reply:
The `site-controller` creates, for example, secrets (certs, keys) used by the `router`.

Reply:
it would feel a tiny bit more comfortable if you we could check that those resources don't have `qontract` annotations on them, so that we don't accidentally delete things just because of naming.

Reply:
Good point! I'll add a check for that.